Userland/Applications/Mail/MailWidget.cpp

```diff
@@ -126,12 +126,15 @@ bool MailWidget::connect_and_login()
             return false;
     }
 
-    m_imap_client = make<IMAP::Client>(server, port, tls);
-    auto connection_promise = m_imap_client->connect();
-    if (!connection_promise) {
-        GUI::MessageBox::show_error(window(), String::formatted("Failed to connect to '{}:{}' over {}.", server, port, tls ? "TLS" : "Plaintext"));
+    auto maybe_imap_client = tls ? IMAP::Client::connect_tls(server, port) : IMAP::Client::connect_plaintext(server, port);
+    if (maybe_imap_client.is_error()) {
+        GUI::MessageBox::show_error(window(), String::formatted("Failed to connect to '{}:{}' over {}: {}", server, port, tls ? "TLS" : "Plaintext", maybe_imap_client.error()));
         return false;
     }
+    m_imap_client = maybe_imap_client.release_value();
+
+    auto connection_promise = m_imap_client->connection_promise();
+    VERIFY(connection_promise);
```

Comment:
This isn't very expressive... You're not even `VERIFY()`ing a promise, you're verifying that a pointer to an empty promise isn't null.

Reply:
Made it a little clearer.